File: beacon/web/src/interface/dingtalk.ts
export interface DingTalkSetting {
    id: number
    enabled: boolean
    webhook_masked: string
    webhook_configured: boolean
    secret_configured: boolean
    at_all: boolean
}

export interface DingTalkUpdateArgs {
    enabled: boolean
    webhook: string
    secret: string
    clear_secret: boolean
    at_all: boolean
}

File: beacon/service/schema/dingtalk.go
package schema

type DingTalkSetting struct {
	ID                uint   `json:"id"`
	Enabled           bool   `json:"enabled"`
	WebhookMasked     string `json:"webhook_masked"`
	WebhookConfigured bool   `json:"webhook_configured"`
	SecretConfigured  bool   `json:"secret_configured"`
	AtAll             bool   `json:"at_all"`
}

type DingTalkUpdateArgs struct {
	Enabled     bool   `json:"enabled"`
	Webhook     string `json:"webhook" validate:"omitempty,lte=2048"`
	Secret      string `json:"secret" validate:"omitempty,lte=255"`
	ClearSecret bool   `json:"clear_secret"`
	AtAll       bool   `json:"at_all"`
}
